### ISSUE TITLE
Per-sample phase set information has the wrong type

### ISSUE THREAD

**Issue by nh13** [genome-in-a-bottle/giab_latest_release#15]
Related issue:  https://github.com/brentp/cyvcf2/issues/161.

You can see the original GRCh38 NA12878 VCF below.  You'll see that the type for the `PS` format field is set as `String`, which is not valid according to the VCF 4.2 spec (spec says "Non-negative 32-bit Integer").  This causes errors parsing the VCF when strictly following the spec.  For example `bcftools

<details>

<summary>Getting the original VCF</summary>

```bash
bcftools view https://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/release/NA12878_HG001/latest/GRCh38/HG001_GRCh38_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X-SOLID_CHROM1-X_v.3.3.2_highconf_PGandRTGphasetransfer.vcf.gz chr1:4001310-4001310 | grep PS
##CL=vcffilter -i filtered-phase-transfer.vcf.gz -o - --javascript "ensureFormatHeader(\"##FORMAT=<ID=PS,Number=1,Type=String,Description=\\\"Phase set for GT\\\">\"); function record() {if(INTEGRATION.GT==\"1/1\") { INTEGRATION.IPS=\".\"; INTEGRATION.PS=\"HOMVAR\"; INTEGRATION.GT=\"1|1\";} else {if((INTEGRATION.GT==\"0/1\" || INTEGRATION.GT==\"1/2\" || INTEGRATION.GT==\"2/1\" || INTEGRATION.GT==\"1/0\") ) {if(INTEGRATION.IPS.length>1) {INTEGRATION.PS=INTEGRATION.IPS; INTEGRATION.GT=INTEGRATION.IGT;} else {INTEGRATION.PS=\".\";};} else { if((INTEGRATION.IPS.length<2)) { INTEGRATION.IPS=\".\";} INTEGRATION.PS=\"PATMAT\";};};}"
##INFO=<ID=DPSum,Number=1,Type=Integer,Description="Total read depth summed across all datasets, excluding MQ0 reads">
##FORMAT=<ID=IPS,Number=1,Type=String,Description="Phase set for IGT">
##FORMAT=<ID=PS,Number=1,Type=String,Description="Phase set for GT">
chr1	4001310	.	G	T	50	PASS	platforms=4;platformnames=Illumina,CG,10X,Solid;datasets=4;datasetnames=HiSeqPE300x,CGnormal,10XChromium,SolidSE75bp;callsets=5;callsetnames=HiSeqPE300xGATK,CGnormal,HiSeqPE300xfreebayes,10XGATKhaplo,SolidSE75GATKHC;datasetsmissingcall=IonExome,SolidPE50x50bp;callable=CS_HiSeqPE300xGATK_callable,CS_CGnormal_callable,CS_HiSeqPE300xfreebayes_callable	GT:DP:ADALL:AD:GQ:IGT:IPS:PS	0|1:571:133,124:160,157:1011:0|1:4001310_G_T:PATMAT
```

Tools like `picard FilterVcf` will re-write the VCF header to follow the spec, changing the header line for `PS` to have type `Integer`.

**Comment by yfarjoun** [comment]
:+1:

**Comment by yfarjoun** [comment]
bcftools cannot process these files as seen in this issue: https://github.com/samtools/bcftools/issues/504

**Comment by nh13** [comment]
@yfarjoun bcftools can process it; see my example above “getting the original VCF”. The type is String in the header, which is what the issue you linked suggested.

**Comment by yfarjoun** [comment]
Oh I see. so I need to fix the original VCF before letting gatk see it...thanks!

**Comment by jzook** [comment]
I apologize for my misunderstanding of the VCF spec for PS.  We've fixed this in the v4.2.1 benchmark vcfs we just released for the Ashkenazi trio, and thanks @nh13 for suggesting a fix for our older vcfs!
